toolboxes/mri_core/mri_core_partial_fourier.cpp

```diff
@@ -0,0 +1,613 @@
+            else
+            {
+                startE2 = 0;
+                endE2 = 0;
+            }
+
+            size_t N = dst.get_number_of_elements() / (RO*E1*E2);
+            const T* pSrc = src.begin();
+            T* pDst = dst.begin();
+
+            long long n;
+
+#pragma omp parallel for default(none) private(n) shared(N, pSrc, pDst, RO, E1, E2, startRO, endRO, startE1, endE1, startE2, endE2)
+            for (n = 0; n<(long long)N; n++)
+            {
+                for (size_t e2 = startE2; e2 <= endE2; e2++)
+                {
+                    for (size_t e1 = startE1; e1 <= endE1; e1++)
+                    {
+                        size_t offset = n*RO*E1*E2 + e2*E1*RO + e1*RO + startRO;
+                        memcpy(pDst + offset, pSrc + offset, sizeof(T)*(endRO - startRO + 1));
+                    }
+                }
+            }
+        }
+        catch (...)
+        {
+            GADGET_THROW("Errors happened in partial_fourier_reset_kspace(...) ... ");
+        }
+    }
+
+    // ------------------------------------------------------------------------
+
+    /// create and apply the transition band
+    template <typename T>
+    void partial_fourier_transition_band(const hoNDArray<T>& src, hoNDArray<T>& dst, size_t startRO, size_t endRO, size_t startE1, size_t endE1, size_t startE2, size_t endE2, size_t transBandRO, size_t transBandE1, size_t transBandE2)
+    {
+        try
+        {
+            size_t NDim = src.get_number_of_dimensions();
+            GADGET_CHECK_THROW(NDim >= 3);
+
+            size_t RO = dst.get_size(0);
+            size_t E1 = dst.get_size(1);
+            size_t E2 = dst.get_size(2);
+
+            size_t RO_src = src.get_size(0);
+            size_t E1_src = src.get_size(1);
+            size_t E2_src = src.get_size(2);
+
+            GADGET_CHECK_THROW(RO == RO_src);
+            GADGET_CHECK_THROW(E1 == E1_src);
+            GADGET_CHECK_THROW(E2 == E2_src);
+            GADGET_CHECK_THROW(src.get_number_of_elements() == dst.get_number_of_elements());
+
+            if ((startRO >= RO) || (endRO >= RO) || (startRO>endRO))
+            {
+                dst = src;
+                GWARN_STREAM("partial_fourier_transition_band(...) : (startRO>=RO) || (endRO>=RO) || (startRO>endRO) ... ");
+                return;
+            }
+
+            if ((startE1 >= E1) || (endE1 >= E1) || (startE1>endE1))
+            {
+                dst = src;
+                GWARN_STREAM("partial_fourier_transition_band(...) : (startE1>=E1) || (endE1>=E1) || (startE1>endE1) ... ");
+                return;
+            }
+
+            if ( (E2>1) && ((startE2 >= E2) || (endE2 >= E2) || (startE2>endE2)) )
+            {
+                dst = src;
+                GWARN_STREAM("partial_fourier_transition_band(...) : (startE2>=E2) || (endE2>=E2) || (startE2>endE2) ... ");
+                return;
+            }
+
+            while (transBandRO>1 && startRO + transBandRO > RO / 2)
```

Comment:
Why not do this with min and max operations instead? It's very unclear what's happening here